### PR TITLE
document uri schemes

### PR DESCRIPTION
**Pull request by Simon-Laux**
document uri-schemes that deltachat may do stuff with.
use rich diff or view at https://github.com/deltachat/interface/blob/document_uri_schemes/uri-schemes.md

### CODE REVIEW

File: uri-schemes.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# Deltachat URI-Schemes
+
+Currently URI schemes are mainly supported by the app-internal QRcode scanner, at this time only `openpgp4fpr` (used for securejoin) is also registered as uri scheme handler on all platforms ([android](https://github.com/deltachat/deltachat-android), [ios](https://github.com/deltachat/deltachat-ios) and [desktop](https://github.com/deltachat/deltachat-desktop))
+
+- related forum topic: https://support.delta.chat/t/custom-deltachat-url-uri-scheme/346
+- core sourcecode that handles uri schemes: https://github.com/deltachat/deltachat-core-rust/blob/master/src/qr.rs
```
Comment (flub) [inline uri-schemes.md:6]: Also `src/securejoin/qrinvite.rs` if that matters.

File: uri-schemes.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+| Related Terms\*         | Securejoin                      |
+| Availible on            | iOS, android, desktop           |
+| Decoded by the core\*\* | Yes                             |
+| Other apps using it     | <li>OpenKeychain(android)</li>  |
+
+### Syntax
+
+```
+OPENPGP4FPR:FINGERPRINT#a=ADDR&n=NAME&i=INVITENUMBER&s=AUTH
+OPENPGP4FPR:FINGERPRINT#a=ADDR&g=GROUPNAME&x=GROUPID&i=INVITENUMBER&s=AUTH
```
Comment (flub) [inline uri-schemes.md:51]: a, g and n are URL-encoded, not sure if you need to point this out.
i & s are 66 random bits endcoded as base-64.

File: uri-schemes.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+OPENPGP4FPR:EEA98F87742EF2FD6C23677F1E1142828C202998#a=demo.fn8hk%40five.chat&n=&i=rd82URz8_ac&s=MFRLUHvIHlq
+```
+
+join regular group
+
+```
+OPENPGP4FPR:EEA98F87742EF2FD6C23677F1E1142828C202998#a=demo.fn8hk%40five.chat&g=groupname2&x=iNekF7uW509&i=xW0ZAcLUQDf&s=OYV-PWe63Vv
+```
+
+join verified/protected group
```
Comment (flub) [inline uri-schemes.md:68]: maybe point out that this is the same as joining a regular group?  you can't distinguish on the QR code afaik

**Review by flub — APPROVED** [review]
I only reviewed the securejoin ones

File: uri-schemes.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+We plan to add an option to specify the credentails in the url in the future.
+
+## **DCWEBRTC**
+
+|                          |                                                          |
+| ------------------------ | -------------------------------------------------------- |
+| Scheme                   | `DCWEBRTC:`                                              |
+| Used for                 | Setting the `WebrtcInstance` config value from an qrcode |
+| Related Terms\*          | webrtc_instance                                          |
+| Availible on             | ? [TODO]                                                 |
```
Comment (link2xt) [inline uri-schemes.md:106]: "Available"

File: uri-schemes.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+```
+VCARD:BEGIN\nN:last name;first name;...;\nEMAIL;<type>:addr...;
+```
+
+## URI schemes that are qrcode reader only
+
+- `http://` & `https://` - to make the qr code scanner more complete and allow users to scan common qr codes.
+
+## Registered Handelers per Platform
```
Comment (link2xt) [inline uri-schemes.md:218]: "Handlers"

File: uri-schemes.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+| Used for | oauth2 Login (only android currently) |
+| Related Terms\* | - |
+| Availible on | iOS (defined, but not used yet), android |
+| Handled by the core| No|
+| Other apps using it | none, only DeltaChat |
+| Notes | Only used as scheme currently, not in qr codes |
+
+### Syntax
+
+[TODO]
```
Comment (link2xt) [inline uri-schemes.md:23]: What this section is for?

File: uri-schemes.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+| Other apps using it     | <li>OpenKeychain(android)</li>  |
+
+### Syntax
+
+```
+OPENPGP4FPR:FINGERPRINT#a=ADDR&n=NAME&i=INVITENUMBER&s=AUTH
+OPENPGP4FPR:FINGERPRINT#a=ADDR&g=GROUPNAME&x=GROUPID&i=INVITENUMBER&s=AUTH
+```
+
+The fields `a`, `g` and `n` are URL encoded.
```
Comment (link2xt) [inline uri-schemes.md:54]: Maybe note that everything after `#` is a Delta Chat extension.

**Review by link2xt — APPROVED** [review]

File: uri-schemes.md (hunk 6 above)
Comment (Simon-Laux) [reply uri-schemes.md:23]: what the parts an uri with that uri scheme is made of

**Pull request merged**